helm/chart/router/templates/secret.yaml

```diff
@@ -1,10 +1,10 @@
-{{- if not (empty .Values.managedFederation.apiKey) }}
+{{- if (or (.Values.managedFederation.apiKeyExistingSecret) (.Values.managedFederation.apiKey)) }}
```

Comment:
Don't we only want to create this secret if the apiKey is specified? It could become a source of confusion if it exists but isn't used.

Reply:
Oh yeah, you're totally right